contrib/bindings/go/pathrs/pathrs.go

```diff
@@ -0,0 +1,239 @@
+// with attributes mode and dev
+func (r *Root) Mknod(path string, mode uint, dev int) error {
+	status := mknod(r.root, path, mode, dev)
+	if status < 0 {
+		return handleErr(pointerOnRoot, r.root)
+	}
+
+	return nil
+}
+
+// Hardlink creates a hardlink of file named target and place it to path
+func (r *Root) Hardlink(path, target string) error {
+	status := hardlink(r.root, path, target)
+	if status < 0 {
+		return handleErr(pointerOnRoot, r.root)
+	}
+
+	return nil
+}
+
+// Symlink creates a symlink of file named target and place it to path
+func (r *Root) Symlink(path, target string) error {
+	status := symlink(r.root, path, target)
+	if status < 0 {
+		return handleErr(pointerOnRoot, r.root)
+	}
+
+	return nil
+}
+
+// Free frees underling caught resources
+func (r *Root) Free() {
+	/*
+		Should we handle it in this way or always return not nil pointer on root?
+	*/
+	if r != nil {
+		free(pointerOnRoot, r.root)
+	}
+}
+
+// Handle represents an handle api of rspath
+type Handle struct {
+	handle cRootPtr
+}
+
+// Free frees underling caught resources
+func (h *Handle) Free() {
+	if h == nil {
+		return
+	}
+
+	free(pointerOnHandle, h.handle)
+}
+
+// Reopen upgrade the handle to a file representation
+func (h *Handle) Reopen(flags int) (*os.File, error) {
+	fd := reopen(h.handle, flags)
+	// TODO: verify what is correct path to handle this type of errors
+	// must it always be called error handling as in cat.c file or we can check fd first
+	if err := handleErr(pointerOnHandle, h.handle); err != nil {
+		return nil, err
+	}
+
+	file := os.NewFile(uintptr(fd), "")
+	return file, nil
+}
+
+// Error representation of rust error
+// particularly useful to not frighten to lost controll of pointer which can be rewritten.
+type Error struct {
+	description string
+	backtrace   []backtraceLine
+}
+
+type backtraceLine struct {
+	ip       uintptr
+	sAddress uintptr
+	sName    string
+	sFile    string
+	sLineno  uint32
+}
+
+func (err *Error) Error() string {
+	return err.description
+}
+
+// Backtrace flush backtrace of underlying error to string.
+//
+// Its not passed to realization of Error interface on purpose since
+// the main error should remain clear and simple
+func (err *Error) Backtrace() string {
+	buf := strings.Builder{}
+	buf.Reset()
+
+	for _, line := range err.backtrace {
+		if line.sName != "" {
+			buf.WriteString(fmt.Sprintf("'%s'@", line.sName))
+		}
+		buf.WriteString(fmt.Sprintf("<0x%x>+0x%x\n", line.sAddress, uintptr(line.ip)-uintptr(line.sAddress)))
+		if line.sFile != "" {
+			buf.WriteString(fmt.Sprintf("  in file '%s':%d\n", line.sFile, line.sLineno))
+		}
+	}
+
+	return buf.String()
+}
+
+func newError(err cErrorPtr) *Error {
+	var description string
+	var backtrace []backtraceLine
+
+	realError := (*errorType)(err)
+
+	if realError.savedErrno != 0 {
+		description = fmt.Sprint(syscall.Errno(realError.savedErrno))
+	} else if realError.description != nil {
+		description = toGoString(realError.description)
+	}
```

Comment:
The description includes more information than just the errno. Really the way this should be done is always storing `description`, and if the `saved_errno` is non-zero then you add a wrapped `syscall.Errno` which is returned on `Unwrap`.